extensions/amp-a4a/0.1/real-time-config-manager.js

```diff
@@ -65,331 +70,376 @@ export const RTC_ERROR_ENUM = {
           checkStillCurrent();
-          const rtcTime = Date.now() - rtcStartTime;
-          // An empty text response is allowed, not an error.
-          if (!text) {
-            return {rtcTime, callout};
-          }
-          const response = tryParseJson(text);
-          return response ? {response, rtcTime, callout} :
-            buildErrorResponse_(
-                RTC_ERROR_ENUM.MALFORMED_JSON_RESPONSE, callout,
-                errorReportingUrl, win, ampDoc, rtcTime);
-        });
-      })).catch(error => {
-    return error.message == 'CANCELLED' ? undefined : buildErrorResponse_(
-        // The relevant error message for timeout looks like it is
-        // just 'message' but is in fact 'messageXXX' where the
-        // X's are hidden special characters. That's why we use
-        // match here.
-        (error.message && error.message.match(/^timeout/)) ?
-          RTC_ERROR_ENUM.TIMEOUT : RTC_ERROR_ENUM.NETWORK_FAILURE,
-        callout, errorReportingUrl, win, ampDoc, Date.now() - rtcStartTime);
-  });
-}
-
-/**
- * Attempts to parse the publisher-defined RTC config off the amp-ad
- * element, then validates that the rtcConfig exists, and contains
- * an entry for either vendor URLs, or publisher-defined URLs. If the
- * config contains an entry for timeoutMillis, validates that it is a
- * number, or converts to a number if number-like, otherwise overwrites
- * with the default.
- * IMPORTANT: If the rtcConfig is invalid, RTC is aborted, and the ad
- *   request continues without RTC.
- * @param {!Element} element
- * @return {?Object}
- * @visibleForTesting
- */
-export function validateRtcConfig_(element) {
-  const defaultTimeoutMillis = 1000;
-  const unparsedRtcConfig = element.getAttribute('rtc-config');
-  if (!unparsedRtcConfig) {
-    return null;
-  }
-  const rtcConfig = tryParseJson(unparsedRtcConfig);
-  if (!rtcConfig) {
-    user().warn(TAG, 'Could not parse rtc-config attribute');
-    return null;
+          return res.text().then(text => {
+            checkStillCurrent();
+            const rtcTime = Date.now() - this.rtcStartTime_;
+            // An empty text response is allowed, not an error.
+            if (!text) {
+              return {rtcTime, callout};
+            }
+            const response = tryParseJson(text);
+            return response ? {response, rtcTime, callout} :
+              this.buildErrorResponse_(
+                  RTC_ERROR_ENUM.MALFORMED_JSON_RESPONSE, callout,
+                  errorReportingUrl, rtcTime);
+          });
+        })).catch(error => {
+      return error.message == 'CANCELLED' ? undefined :
+        this.buildErrorResponse_(
+            // The relevant error message for timeout looks like it is
+            // just 'message' but is in fact 'messageXXX' where the
+            // X's are hidden special characters. That's why we use
+            // match here.
+            (error.message && error.message.match(/^timeout/)) ?
```

Comment:
If you use test you don't need to check error.message existence as null|undefined won't match:  /^timeout/.test(error.message)